music_assistant/controllers/streams.py

```diff
@@ -986,19 +986,21 @@ async def get_plugin_source_stream(
         player = self.mass.players.get(player_id)
         plugin_prov: PluginProvider = self.mass.get_provider(plugin_source_id)
         plugin_source = plugin_prov.get_source()
-        if plugin_source.in_use_by and plugin_source.in_use_by != player_id:
-            raise RuntimeError(
-                f"PluginSource plugin_source.name is already in use by {plugin_source.in_use_by}"
-            )
-        self.logger.debug("Start streaming PluginSource %s to %s", plugin_source_id, player_id)
-        audio_input = (
-            plugin_prov.get_audio_stream(player_id)
-            if plugin_source.stream_type == StreamType.CUSTOM
-            else plugin_source.path
-        )
-        player.active_source = plugin_source_id
-        plugin_source.in_use_by = player_id
+        error = None
         try:
+            if plugin_source.in_use_by and plugin_source.in_use_by != player_id:
+                error = RuntimeError(
+                    f"PluginSource plugin_source.name is already in use by {plugin_source.in_use_by}"
+                )
+                raise error
+            self.logger.debug("Start streaming PluginSource %s to %s", plugin_source_id, player_id)
+            audio_input = (
+                plugin_prov.get_audio_stream(player_id)
+                if plugin_source.stream_type == StreamType.CUSTOM
+                else plugin_source.path
+            )
+            player.active_source = plugin_source_id
```

Comment:
this is actually not a setter, so a bug in current dev branch after the player controller/model refactor